flask_sqlalchemy/__init__.py

```diff
@@ -515,9 +515,14 @@ def __get__(self, obj, type):
         try:
             mapper = orm.class_mapper(type)
             if mapper:
-                return type.query_class(mapper, session=self.sa.session())
+                query = type.query_class(mapper, session=self.sa.session())
         except UnmappedClassError:
-            return None
+            query = None
+
+        if obj is not None and not obj.__class__.__bases__[0].query:
```

Comment:
I'm not sure what the second part of this check is doing. Please add a test covering it.

Is there a reason this has to come after the try block?